flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogStateBackend.java

```diff
@@ -272,7 +273,16 @@ public StateBackend configure(ReadableConfig config, ClassLoader classLoader)
                                         : new ChangelogStateBackendHandleImpl(
                                                 singletonList(keyedStateHandle),
                                                 emptyList(),
-                                                keyedStateHandle.getKeyGroupRange()))
+                                                keyedStateHandle.getKeyGroupRange(),
+                                                getMaterializationID(keyedStateHandle)))
                 .collect(Collectors.toList());
     }
+
+    private long getMaterializationID(KeyedStateHandle keyedStateHandle) {
+        if (keyedStateHandle instanceof IncrementalKeyedStateHandle) {
+            return ((IncrementalKeyedStateHandle) keyedStateHandle).getCheckpointId();
```

Comment:
I think we can introduce a new interface to include `getCheckpointId()` instead of still use the `IncrementalKeyedStateHandle`, so that all 3rd-party state handles could extend that handle to make it compatiable with changelog state-backend.

Reply:
WDYT about `CheckpointBoundKeyedStateHandle` name for this interface?

Reply:
How about a straightforward `CheckpointStateHandle`?

Reply:
I've added `CheckpointBoundKeyedStateHandle` as agreed offline, PTAL.